### PR TITLE
Small code quality improvements for the UnitTests

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This **PR** fixes some code quality issues with the unit tests:

- Add additional test categories (traits), to be able to filter tests by
- Fix namespaces to match file locations
- Add missing usings / dispose
- Fix some Warnings

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
@brianpopow  Love that you're doing this!

One thing we're capable of doing now we've inlined the ref images is refactoring the tests to use the same directory and namespacing structure as the src. Currently it's really hard to find where tests are. 

Big job though, for another time...

**Comment by brianpopow** [comment]
> One thing we're capable of doing now we've inlined the ref images is refactoring the tests to use the same directory and namespacing structure as the src. Currently it's really hard to find where tests are.
> 
> Big job though, for another time...

Sounds like a good idea, I will put it on my TODO list.

File: tests/ImageSharp.Tests/Formats/Png/PngMetadataTests.cs [hunk 1]
```diff
@@ -277,20 +277,20 @@ private static void VerifyExifDataIsPresent(ExifProfile exif)
...
-            Assert.Contains(meta.TextData, m => m.Keyword.Equals("International2") && m.Value.Equals("ИМАГЕШАРП") && m.LanguageTag.Equals("rus"));
-            Assert.Contains(meta.TextData, m => m.Keyword.Equals("CompressedInternational") && m.Value.Equals("la plume de la mante") &&
-                                                m.LanguageTag.Equals("fra") && m.TranslatedKeyword.Equals("foobar"));
-            Assert.Contains(meta.TextData, m => m.Keyword.Equals("CompressedInternational2") && m.Value.Equals("這是一個考驗") &&
-                                                m.LanguageTag.Equals("chinese"));
-            Assert.Contains(meta.TextData, m => m.Keyword.Equals("NoLang") && m.Value.Equals("this text chunk is missing a language tag"));
-            Assert.Contains(meta.TextData, m => m.Keyword.Equals("NoTranslatedKeyword") && m.Value.Equals("dieser chunk hat kein übersetztes Schlüßelwort"));
+            Assert.Contains(meta.TextData, m => m.Keyword.Equals("Comment", System.StringComparison.Ordinal) && m.Value.Equals("comment", System.StringComparison.Ordinal));
```
Comment (iamcarbon) [inline tests/ImageSharp.Tests/Formats/Png/PngMetadataTests.cs:280]: Can also use pattern matching against a literal and do m.Keyword is "Comment"

This is already ordinal.

File: tests/ImageSharp.Tests/Formats/Png/PngMetadataTests.cs [hunk 2]
```diff
@@ -277,20 +277,20 @@ private static void VerifyExifDataIsPresent(ExifProfile exif)
...
+            Assert.Contains(meta.TextData, m => m.Keyword.Equals("Author", System.StringComparison.Ordinal) && m.Value.Equals("ImageSharp", System.StringComparison.Ordinal));
```
Comment (iamcarbon) [inline tests/ImageSharp.Tests/Formats/Png/PngMetadataTests.cs:281]: Suggested change: m.Keyword is "Author"

File: tests/ImageSharp.Tests/Formats/Png/PngMetadataTests.cs (hunk 1 above)
Comment (brianpopow) [reply tests/ImageSharp.Tests/Formats/Png/PngMetadataTests.cs]: thanks, that was new to me.

File: tests/ImageSharp.Tests/Processing/Transforms/PadTest.cs [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 namespace SixLabors.ImageSharp.Tests.Processing.Transforms
 {
+    [Trait("Category", "Processors")]
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/Processing/Transforms/PadTest.cs:11]: Might be an idea in the future to create dedicated attributes for these kind of repetitive values.

**Review by JimBobSquarePants — APPROVED** [review]
Really great stuff! Was surprised by how much leaking was going on too! 😲

**Pull request merged**